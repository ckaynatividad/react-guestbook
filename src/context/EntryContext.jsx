import { createContext, useContext, useState } from 'react';

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);

  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};

const useEntries = () => {
  const context = useContext(EntryContext);

  if (context === undefined) {
    throw new Error('must be defined with entryprovider');
  }
  return context;
};

export { EntryProvider, useEntries };
