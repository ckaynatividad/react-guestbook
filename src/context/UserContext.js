import { createContext, useContext, useMemo, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  console.log(process.env.AUTH_EMAIL);
  const login = (email, password) => {
    const loginSuccessful =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ email });
    return loginSuccessful;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  const value = { user, setUser, login, logout };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};
const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('must be used within userprovider');
  }
  return context;
};

export { UserProvider, useUser };
