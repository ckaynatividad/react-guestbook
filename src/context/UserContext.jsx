import { createContext, useContext, useMemo, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};
const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('must be used within userprovider');
  }
  return context;
};

export { UserProvider, useUser };
