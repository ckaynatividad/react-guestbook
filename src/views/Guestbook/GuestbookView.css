.entries {
  width: 300px;
  margin-left: 60%;
  animation: example2 3s ease-out;
}
/* Move it (define the animation) */
@keyframes example2 {
  0% {
    transform: translateY(200%);
  }
  100% {
    transform: translateY(0%);
  }
}

