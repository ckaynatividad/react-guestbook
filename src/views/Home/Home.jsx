import React from 'react';
import Slideshow from '../../components/Main/Slideshow';

export default function Home() {
  return (
    <div>
      <Slideshow />
    </div>
  );
}
