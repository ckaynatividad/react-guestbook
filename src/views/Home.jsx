import Entries from '../components/Entries';
import GuestBook from '../components/GuestBook';
import { Header } from '../components/Header';
import { EntryProvider } from '../context/EntryContext';
//import { EntryProvider } from '../context/EntryContext';

export const Home = () => {
  return (
    <EntryProvider>
      <Header />
      <GuestBook />
      <Entries />
    </EntryProvider>
  );
};
