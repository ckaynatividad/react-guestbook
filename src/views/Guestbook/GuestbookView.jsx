import { NavLink } from 'react-router-dom';
import Entries from '../../components/Guestbook/Entries';
import GuestBook from '../../components/Guestbook/GuestBook';

import { EntryProvider } from '../../context/EntryContext';
import './GuestbookView.css';

export const GuestbookView = () => {
  return (
    <EntryProvider>
      <div className="meow">
        <GuestBook />
      </div>

      <div className="entries">
        <Entries />
      </div>
    </EntryProvider>
  );
};
