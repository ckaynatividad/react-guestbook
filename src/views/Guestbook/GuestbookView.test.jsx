import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { EntryProvider } from '../../context/EntryContext';
import { UserProvider } from '../../context/UserContext';
import { GuestbookView } from './GuestbookView';
import userEvent from '@testing-library/user-event';
import Home from '../Home/Home';

test('renders guestbooks in app', () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <GuestbookView />
      </UserProvider>
    </MemoryRouter>
  );

  const header = screen.getByText(/Welcome to Meow's and Meow's Guestbook/i);
  expect(header).toBeInTheDocument();

  const input = screen.getByLabelText(/Your Entry/i);
  expect(input).toBeInTheDocument();
});

test('renders signout', () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <GuestbookView />
      </UserProvider>
    </MemoryRouter>
  );
  const name = screen.getByLabelText(/Name/i);
  userEvent.type(name, 'meow');

  const entry = screen.getByLabelText(/Your Entry/i);
  userEvent.type(entry, 'meows entry');

  const button = screen.getByText('Submit');
  userEvent.click(button);

  const signOut = screen.getByText('Not meow?');
  expect(signOut).toBeInTheDocument();
});

test('guestbook works', () => {
  const container = render(
    <MemoryRouter>
      <UserProvider>
        <EntryProvider>
          <GuestbookView name="meow" entry="meows entry" />
        </EntryProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});

test('goes back home', () => {
  render(
    <MemoryRouter initialEntries={['/guestbook']}>
      <UserProvider>
        <Route path="/guestbook">
          <GuestbookView />
        </Route>
      </UserProvider>
    </MemoryRouter>
  );
  const back = screen.getByText('go back');
  userEvent.click(back);
  render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );

  const header = screen.getByText(/Welcome to Meow Meow Wedding Planning/i);
  expect(header).toBeInTheDocument();
});
