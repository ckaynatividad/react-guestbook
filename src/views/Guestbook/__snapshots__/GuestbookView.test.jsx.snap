// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`guestbook works 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="mode-button"
      >
        Switch Mode
      </button>
      <p>
        <a
          href="/home"
        >
          go back
        </a>
      </p>
      <header>
        <h2>
          Welcome to Meow's and Meow's Guestbook
        </h2>
      </header>
      <div
        class="fade"
      >
        <div
          class="meow"
        >
          <div
            class="form"
          >
            <form>
              <label>
                Your Entry: 
                <input
                  class="yeah"
                  id="entry-box"
                  type="text"
                  value=""
                />
              </label>
              <p>
                <button
                  class="submit"
                  type="submit"
                >
                  Submit
                </button>
              </p>
            </form>
          </div>
        </div>
        <div
          class="entries"
        >
          <div>
            <h2>
              All Entries
            </h2>
            <ul>
              <p
                class="msg"
              >
                "We love you and are rooting for you."
              </p>
              <p
                class="name"
              >
                - Mom and Dad
              </p>
              <p
                class="msg"
              >
                "Meow meow meow"
              </p>
              <p
                class="name"
              >
                - Meow
              </p>
            </ul>
            <ul
              class="list"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="mode-button"
    >
      Switch Mode
    </button>
    <p>
      <a
        href="/home"
      >
        go back
      </a>
    </p>
    <header>
      <h2>
        Welcome to Meow's and Meow's Guestbook
      </h2>
    </header>
    <div
      class="fade"
    >
      <div
        class="meow"
      >
        <div
          class="form"
        >
          <form>
            <label>
              Your Entry: 
              <input
                class="yeah"
                id="entry-box"
                type="text"
                value=""
              />
            </label>
            <p>
              <button
                class="submit"
                type="submit"
              >
                Submit
              </button>
            </p>
          </form>
        </div>
      </div>
      <div
        class="entries"
      >
        <div>
          <h2>
            All Entries
          </h2>
          <ul>
            <p
              class="msg"
            >
              "We love you and are rooting for you."
            </p>
            <p
              class="name"
            >
              - Mom and Dad
            </p>
            <p
              class="msg"
            >
              "Meow meow meow"
            </p>
            <p
              class="name"
            >
              - Meow
            </p>
          </ul>
          <ul
            class="list"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
